src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,128 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 16;
+    static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private Entry<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = (Entry<K, V>[]) new Entry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if ((float) size / table.length >= LOAD_FACTOR) {
+            resize();
+        }
+        int keyHashcode = key != null ? key.hashCode() : 0;
```

Comment:
Let's move calculate hashcode to separate method.

Reply:
Done